tests/test_compare_dictconfig_vs_dict.py

```diff
@@ -0,0 +1,647 @@
+"""
+This file compares DictConfig methods with the corresponding
+methods of standard python's dict.
+The following methods are compared:
+    __contains__
+    __delitem__
+    __eq__
+    __getitem__
+    __setitem__
+    get
+    pop
+    keys
+    values
+    items
+
+We have separate test classes for the following cases:
+    TestUntypedDictConfig: for DictConfig without a set key_type
+    TestPrimitiveTypeDunderMethods: for DictConfig where key_type is primitive
+    TestEnumTypeDunderMethods: for DictConfig where key_type is Enum
+"""
+from copy import deepcopy
+from enum import Enum
+from typing import Any, Dict, Optional
+
+from pytest import fixture, mark, param, raises
+
+from omegaconf import DictConfig, OmegaConf
+from omegaconf.errors import ConfigKeyError, ConfigTypeError, KeyValidationError
+from tests import Enum1
+
+
+@fixture(
+    params=[
+        "str",
+        1,
+        3.1415,
+        True,
+        Enum1.FOO,
+    ]
+)
+def key(request: Any) -> Any:
+    """A key to test indexing into DictConfig."""
+    return request.param
+
+
+@fixture
+def python_dict(data: Dict[Any, Any]) -> Dict[Any, Any]:
+    """Just a standard python dictionary, to be used in comparison with DictConfig."""
+    return deepcopy(data)
+
+
+@fixture(params=[True, False, None])
+def struct_mode(request: Any) -> Optional[bool]:
+    struct_mode: Optional[bool] = request.param
+    return struct_mode
```

Comment:
I am not sure we need to test None because at least for Struct Mode, it supposed to behave like False.
Did it catch anything interesting?

In terms of order: I would go with `[None, False, True]`.

Reply:
> I am not sure we need to test None because at least for Struct Mode, it supposed to behave like False.
> Did it catch anything interesting?

No, nothing interesting.

I did notice that DictConfig.pop and DictConfig.__delitem__ have some special logic for if `struct_mode is False`: if `cfg._is_typed()` is `True`, i.e. if `object_type` is set, then `struct_mode` must be `False` for `__delitem__` or `pop` to work. If `struct_mode` is `None`, then `__delitem__` and `pop` raise `ConfigTypeError`. This use case does not come up in the tests from this file.

> In terms of order: I would go with `[None, False, True]`.

Updated to `[None, False, True]` in https://github.com/omry/omegaconf/pull/484/commits/baa918c022969dd8960e9a41f7aadee10b12a60e